playback/src/audio_backend/gstreamer.rs

```diff
@@ -0,0 +1,115 @@
+use super::{Open, Sink};
+use gst::prelude::*;
+use gst::*;
+use std::sync::mpsc::{sync_channel, SyncSender};
+use std::{io, thread};
+use zerocopy::*;
+
+#[allow(dead_code)]
+pub struct GstreamerSink {
+    tx: SyncSender<Vec<u8>>,
+    pipeline: gst::Pipeline,
+}
+
+impl Open for GstreamerSink {
+    fn open(device: Option<String>) -> GstreamerSink {
+        gst::init().expect("Failed to init gstreamer!");
+        let pipeline_str_preamble = r#"appsrc caps="audio/x-raw,format=S16LE,layout=interleaved,channels=2,rate=44100" block=true max-bytes=4096 name=appsrc0 "#;
+        let pipeline_str_rest = r#" ! audioconvert ! autoaudiosink"#;
+        let pipeline_str: String = match device {
+            Some(x) => format!("{}{}", pipeline_str_preamble, x),
```

Comment:
I love this idea, insanely flexible - but dead simple (for people familiar with gstreamer anyway).